source/ada/lsp-ada_handlers.adb

```diff
@@ -1716,6 +1717,29 @@ package body LSP.Ada_Handlers is
             and then Node.Parent.Parent.Kind in Ada_Derived_Type_Def_Range));
       --  Return True if the node belongs to derived type declaration.
 
+      function Is_Access_Ref (Node : Ada_Node) return Boolean;
+
+      -------------------
+      -- Is_Access_Ref --
+      -------------------
+
+      function Is_Access_Ref (Node : Ada_Node) return Boolean is
+      begin
+         if not Node.Parent.Is_Null and then Node.Parent.Kind in Ada_Name then
+            declare
+               Sibling : constant Ada_Node := Node.Next_Sibling;
+               Text    : constant Wide_Wide_String :=
+                 (if Sibling.Is_Null
+                  then ""
+                  else Ada.Wide_Wide_Characters.Handling.To_Lower
+                    (Sibling.Text));
+            begin
+               return Text = "access" or else Text = "unrestricted_access";
```

Comment:
What's about `X'Unchecked_Access`? Do we need it here? May be `X'Address` also?

Reply:
Unchecked_Access looks like a good idea.
For Address looking at the description "Address may be used to return the address of the first element allocated to X", it seems out of scope. If you disagree, I can add it: I have no strong opinion about it.

Reply:
I think that you can add 'Address too, it's used sometimes in bare-metal programming so that will be useful